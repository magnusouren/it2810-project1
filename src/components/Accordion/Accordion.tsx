import React from 'react';

type AccordionProps = {
  title: string;
  children: React.ReactNode;
};
const Accordion = (props: AccordionProps) => {
  const { title, children } = props;

  return (
    <div className='accordion'>
      <h3 className='accordion-title'>{title}</h3>
      <div className='accordion-content'>{children}</div>
    </div>
  );
};

export default Accordion;
