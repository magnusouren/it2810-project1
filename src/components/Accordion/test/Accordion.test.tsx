import { render, screen } from '@testing-library/react';
import { describe, it } from 'vitest';

import Accordion from '../Accordion';

// Les mer: https://eternaldev.com/blog/testing-a-react-application-with-vitest

describe('Accordion test', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <Accordion title='Testing'>
        <h4>Content</h4>
      </Accordion>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should show title', () => {
    render(
      <Accordion title='Testing'>
        <h4>Content</h4>
      </Accordion>,
    );

    expect(screen.getByText(/Testing/i)).toBeDefined();
    expect(screen.getByRole('heading', { name: /Testing/i })).toBeDefined();
  });
});
