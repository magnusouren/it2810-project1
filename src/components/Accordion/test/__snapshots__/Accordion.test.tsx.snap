// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion test > Should match snapshot 1`] = `
<div>
  <div
    class="accordion"
  >
    <h3
      class="accordion-title"
    >
      Testing
    </h3>
    <div
      class="accordion-content"
    >
      <h4>
        Content
      </h4>
    </div>
  </div>
</div>
`;
